### PR TITLE
[flann] Change from find_package(hdf5) to find_dependency(HDF5)

### PR DESCRIPTION
**Pull request by longhuan2018**
fix #22153

### CODE REVIEW

**Review by BillyONeal — APPROVED** [review]

**Comment by JackBoosY** [comment]
Will rerun test after x64-windows finished.
Also please report this to the upstream.

Thanks.

File: ports/flann/portfile.cmake [hunk 1]
```diff
@@ -1,14 +1,20 @@
 #the port uses inside the CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS, which is discouraged by vcpkg.
 
+set(FLANN_PATCHES
+    fix-build-error.patch
+)
+
+if("hdf5" IN_LIST FEATURES)
+    list(APPEND FLANN_PATCHES fix-dependency-hdf5.patch)
+endif()
```
Comment (dg0yt) [inline ports/flann/portfile.cmake:9]: Please use a non-conditional patch. Either configure the wrapper, or inspect the required libraries.

**Review by dg0yt — CHANGES_REQUESTED** [review]
Please either remove the non-standard `CMAKE_DEBUG_POSTFIX`, or move it to `OPTIONS_DEBUG`.

And check if other build warnings that can be resolved.

File: ports/flann/portfile.cmake (hunk 1 above)
Comment (JackBoosY) [reply ports/flann/portfile.cmake]: Yes, it may affect other patches.

**Review by JackBoosY — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks <3

**Pull request merged**